mycroft/skills/skill_loader.py

```diff
@@ -123,11 +124,58 @@ def _get_last_modified_time(path):
         return 0
 
 
+def is_skill_v2(module):
```

Comment:
Another versioning thing to track?  Is the "skill framework version" official?  If so, where is it documented?  Is there a way around this?  Will we just keep adding versions upon versions?  When do old versions get cleaned up?

Reply:
The current version is referred to v2 in the skill loader.

We can't drop support for all existing skills without having an alternative in place, so we need to support the current version for some time when making changes. I don't think that more than two versions should ever be supported at the same time. Likely the v2 should be kept for a year or so?

Reply:
Oh also as touched on above, we can probably skip the compatibility checks if we do some hacking in `__new__()` buuut that's a slippery-slope that I'd rather avoid.